samples/webforms/integrate-report-web-app/PBIWebApp/Default.aspx

```diff
@@ -1,52 +1,109 @@
 
         //This code is for sample purposes only.
 
         //Configure IFrame for the Report after you have an Access Token. See Default.aspx.cs to learn how to get an Access Token
         window.onload = function () {
-            if ("" != document.getElementById('MainContent_accessToken').value)
+            var accessToken = document.getElementById('MainContent_accessToken').value;
+
+            if (!accessToken || accessToken == "")
             {
-                var iframe = document.getElementById('iFrameEmbedReport');
+                return;
+            }
 
-                // To load a Report do the following:
-                // Set the IFrame source to the EmbedUrl from the Get Reports operation
-                iframe.src = document.getElementById('MainContent_ReportEmbedUrl').value;
+            var embedUrl = document.getElementById('MainContent_txtEmbedUrl').value;
+            var reportId = document.getElementById('MainContent_txtReportId').value;
 
-                // Add an onload handler to submit the access token
-                iframe.onload = postActionLoadReport;
-            }
-        };
-        
-        // Post the access token to the IFrame
-        function postActionLoadReport() {
-
-            // Construct the push message structure
-            // this structure also supports setting the reportId, groupId, height, and width.
-            // when using a report in a group, you must provide the groupId on the iFrame SRC
-            var m = {
-                action: "loadReport",
-                accessToken: document.getElementById('MainContent_accessToken').value
+            // Embed configuration used to describe the what and how to embed.
+            // This object is used when calling powerbi.embed.
+            // This also includes settings and options such as filters.
+            // You can find more information at https://github.com/Microsoft/PowerBI-JavaScript/wiki/Embed-Configuration-Details.
+            var config= {
+                type: 'report',
+                accessToken: accessToken,
+                embedUrl: embedUrl,
+                id: reportId,
+                settings: {
+                    filterPaneEnabled: true,
+                    navContentPaneEnabled: true
+                }
             };
-            message = JSON.stringify(m);
 
-            // push the message.
-            iframe = document.getElementById('iFrameEmbedReport');
-            iframe.contentWindow.postMessage(message, "*");;
-        }
+            // Grab the reference to the div HTML element that will host the report.
+            var reportContainer = document.getElementById('reportContainer');
+
+            // Embed the report and display it within the div container.
+            var report = powerbi.embed(reportContainer, config);
+
+            // Report.on will add an event handler which prints to Log window.
+            report.on("loaded", function () {
```

Comment:
What about errors? Noa added detailed error structure ... Worth listening on errors and raising some popup.